packages/vite/src/node/config.ts

```diff
@@ -77,10 +77,8 @@ const promisifiedRealpath = promisify(fs.realpath)
 export interface ConfigEnv {
   command: 'build' | 'serve'
   mode: string
-  /**
-   * @experimental
-   */
-  ssrBuild?: boolean
+  isSsrBuild: boolean
```

Comment:
What is the plan for avoiding #8912? (https://discord.com/channels/804011606160703521/831456449632534538/993429210867707914)
For example, are we going to say "call `loadConfigFromFile` for each build separately"?
Otherwise, I guess vitepress won't be able to pass the correct value.

Reply:
The idea I'm thinking is that, if you don't know or can't be sure what it is, you can simply set `false` (we do the same for `vitestSetup.ts` in this PR). Maybe it makes more sense to allow `undefined` to convey the "i don't know part"?

Reply:
I've pushed a commit that makes them `undefined`-able